Libs/MRML/Core/vtkMRMLI18N.cxx

```diff
@@ -147,3 +147,45 @@ std::string vtkMRMLI18N::Translate(const char *context, const char *sourceText,
     return sourceText ? sourceText : "";
   }
 }
+
+//----------------------------------------------------------------------------
+std::string vtkMRMLI18N::Format(const std::string& input,
+  const char* arg1/*=nullptr*/, const char* arg2/*=nullptr*/, const char* arg3/*=nullptr*/,
+  const char* arg4/*=nullptr*/, const char* arg5/*=nullptr*/, const char* arg6/*=nullptr*/,
+  const char* arg7/*=nullptr*/, const char* arg8/*=nullptr*/, const char* arg9/*=nullptr*/)
+{
+  std::string output;
+
+  for (std::string::const_iterator it = input.cbegin(); it != input.cend(); ++it)
+  {
+    if ((*it == '%') && (it + 1 != input.end()))
+    {
+      const char nextChar = *(it + 1);
+      if (nextChar == '%')
+      {
+        // Escaped '%'
+        output += "%";
+        ++it;
+        continue;
+      }
+      else if (std::isdigit(nextChar))
+      {
+        // Found a valid placeholder (e.g., %1, %2, %3, ..., %9)
+        if (nextChar == '1' && arg1) { output += arg1; }
```

Comment:
Maybe the using like below is better, to avoid multiple times of arg1 check and if-else blocks
if (arg1) {
switch (nextChar) {
case '1': ... break;
case '2' ... break;
}
}